src/rm/rm.rs

```diff
@@ -129,7 +129,7 @@ fn remove(files: Vec<String>, force: bool, interactive: InteractiveMode, one_fs:
         let filename = &filename[..];
         let file = Path::new(filename);
         if file.exists() {
-            if file.is_dir() {
+            if file.symlink_metadata().unwrap().is_dir() {
```

Comment:
That may be the case, but the documentation mentions that it may also fail if you don't have the correct permissions (although I'm not totally sure when this would occur).